corehq/apps/userreports/views.py

```diff
@@ -515,14 +521,22 @@ def _get_config_kwargs(self, app_source):
             )
         }
 
-    def _build_data_source(self, app_source, username):
+    def _expire_data_source(self, data_source_config_id):
+        with temp_setting_value('CELERY_ALWAYS_EAGER', False):  # so it works in dev environments too
+            delete_data_source_task.apply_async(
+                (self.domain, data_source_config_id),
+                countdown=TEMP_DATA_SOURCE_LIFESPAN
+            )
```

Comment:
This is going to kill the data source used by the report builder for previewing (not the data source created when the report is saved). It assumes that a user is not going to take longer than 24 hours to create a report. @NoahCarnahan @sheelio do you think that's a safe assumption?